src/openforms/submissions/api/viewsets.py

```diff
@@ -297,25 +296,61 @@ def _complete(self, request, *args, **kwargs):
         status endpoint that a retry is needed, the ID is added back to the session.
 
         ---
-        **Warning**
+        **Validation errors**
+
+        The validation errors are returned in the usual `invalidParams` structure. The
+        following errors can occur:
+
+        * name `privacyPolicyAccepted` - unchecked while accepting it is required
+        * name `statementOfTruthAccepted` - unchecked while accepting it is required
+        * name `steps[i].nonFieldErrors`:
+
+            * code `blocked` - a logic rule prevents the step from being submitted,
+              which blocks the submission as a whole from being completed.
+            * code `incomplete` - there is no or incomplete step data submitted for the
+              step at this index.
+
+        * name `steps[i].data.*` - validation errors related to the formio components
+          in the step data.
+
+        Additionally, if you try to complete a submission that doesn't allow this (due
+        to form-level configuration), you will get an HTTP 403 error.
 
-        The schema of the validation errors response is currently marked as
-        experimental. See our versioning policy in the developer documentation for
-        what this means.
         ---
         """
         submission = self.get_object()
+        if submission.form.submission_allowed != SubmissionAllowedChoices.yes:
+            raise PermissionDenied(
+                detail=_("Submission is not enabled for this form."),
```

Comment:
Why did we change the error?I think the old one `Submission of this form is not allowed` makes more sense since the checkbox in the admin declares the same (not required, just an opinion).

Reply:
The biggest problem was that the original problem resulted in an HTTP 400 with validation errors, but this is not something the person filling out the form can correct. IMO this error message makes it more clear that the user cannot correct the problem.